### PR TITLE
Rework: FieldActors

### PR DESCRIPTION
**Pull request by AngeloTadeucci**
- Changed **FieldActors** to be public instead of `private partial`;
- Some minor logic changes to **FieldManager** like using LINQ to check for null objects.

### CODE REVIEW

**Review by Sparkymod — APPROVED** [review]

**Review by Zintixx — APPROVED** [review]

File: MapleServer2/Managers/Actors/FieldActor.cs [hunk 1]
```diff
@@ -5,177 +5,174 @@
...
+    {
+        if (amount <= 0)
         {
-            // TODO: Perform pathfinding, then set path.
-            Velocity = target - Coord;
+            return;
         }
 
-        public virtual void Cast(SkillCast skillCast)
+        lock (Stats)
```
Comment (AlanMorel) [inline MapleServer2/Managers/Actors/FieldActor.cs:69]: Why do we need locks?

Comment (AngeloTadeucci) [reply MapleServer2/Managers/Actors/FieldActor.cs:69]: good question, I believe is to ensure thread safety since Stats is a simple Dictionary

**Review by darkvergus — APPROVED** [review]

File: MapleServer2/Managers/Actors/FieldActor.cs (hunk 1 above)
Comment (AlanMorel) [reply MapleServer2/Managers/Actors/FieldActor.cs:69]: aren't there concurrency-safe dictionaries in C#?

Comment (AngeloTadeucci) [reply MapleServer2/Managers/Actors/FieldActor.cs:69]: There are, I can change it in the next PR

**Pull request merged**